src/main/java/com/gmail/llmdlio/townyflight/TownyFlightAPI.java

```diff
@@ -165,9 +177,12 @@ public void removeFlight(Player player, boolean silent, boolean forced, String c
 	 * @param silent true will mean no message is shown to the {@link Player}.
 	 */
 	public void addFlight(Player player, boolean silent) {
-		if (!silent) Message.of("flightOnMsg").to(player);
-		player.setAllowFlight(true);
-		cachePlayerFlight(player, true);
+		// Added a canFly check
+		if(canFly(player, true)){
+			if (!silent) Message.of("flightOnMsg").to(player);
+			player.setAllowFlight(true);
+			cachePlayerFlight(player, true);
+		}

```

Comment:
This method is not meant to test if a player can fly, it is meant to only give flight.

If you're calling this method in new code, that new method should confirm the player can fly before calling this addFlight method.